sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -741,7 +741,7 @@ private[sql] class SQLConf extends Serializable with CatalystConf with Logging {
 
   def variableSubstituteDepth: Int = getConf(VARIABLE_SUBSTITUTE_DEPTH)
 
-  def warehousePath: String = new Path(getConf(WAREHOUSE_PATH)).toString
+  def warehousePath: String = Utils.resolveURI(getConf(WAREHOUSE_PATH)).toString
```

Comment:
> interpreted as an HDFS path by default because that's not how it worked historically

Except that is how it's worked historically for this piece of code at least. Taking a small detour, the reason I really dislike the use of `resolveURI` everywhere (and I commented on this when that code was added) is that it makes it awkward to use HDFS. Instead of relying on `fs.defaultFS` now you have to know the HDFS scheme (which can be "hdfs", "maprfs", "s3", "webhdfs", "viewfs", or others depending on your particular configuration) or things will break in mysterious ways.

Anyway, I don't really care that much for this particular setting, so if you think changing the behavior is better, it's fine with me.